File: 조준장/level2/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    queue1_sum = sum(queue1)
    queue2_sum = sum(queue2)
    
    q1 = deque(queue1)
    q2 = deque(queue2)
    
    while queue1_sum != queue2_sum:
        if answer > len(queue1) * 4:
            return -1
        
        if queue1_sum > queue2_sum:
            target = q1.popleft()
            queue2_sum += target
            queue1_sum -= target
            q2.append(target)
        else:
            target = q2.popleft()
            queue1_sum += target
            queue2_sum -= target
            q1.append(target)
            
        answer += 1

    
    return answer

File: 조준장/level2/두 큐 합 같게 만들기.kt
class Solution {
    fun solution(queue1: IntArray, queue2: IntArray): Int {
        var answer: Int = 0

        val newQueue1 = ArrayDeque<Long>()
        val newQueue2 = ArrayDeque<Long>()
        
        var sumQueue1 = queue1.sum().toLong()
        var sumQueue2 = queue2.sum().toLong()
        
         for (i in queue1.indices) {
            newQueue1.add(queue1[i].toLong())
            newQueue2.add(queue2[i].toLong())
        }
       
        while (sumQueue1 != sumQueue2) {

            if (answer > queue1.size * 3) {
                return@solution -1
            }
            
            if (sumQueue1 > sumQueue2) {
                val target = newQueue1.removeFirst()
                sumQueue1 -= target
                sumQueue2 += target
                newQueue2.add(target)
                
            } else {
                val target = newQueue2.removeFirst()
                sumQueue2 -= target
                sumQueue1 += target
                newQueue1.add(target)
                
            }
            
            answer += 1
            
        }



        return answer
    }
}
